src/eh1/i2c.rs

```diff
@@ -298,6 +298,34 @@ impl i2c::I2c for Mock {
     }
 }
 
+#[cfg(feature = "embedded-hal-async")]
+impl embedded_hal_async::i2c::I2c for Mock {
+    async fn read(&mut self, address: u8, buffer: &mut [u8]) -> Result<(), Self::Error> {
+        eh1::i2c::I2c::read(self, address, buffer)
```

Comment:
The `eh1` import is aliased as `embedded_hal` further above, so this could be written like this:

```suggestion
        embedded_hal::i2c::I2c::read(self, address, buffer)
```

I like about this variant that it's clearly visible that the `embedded_hal_async` impl calls the `embedded_hal` trait impls.

Another option would be to directly import the `I2c` trait in the imports section, then you could write:

```suggestion
        I2c::read(self, address, buffer)
```

I like about that variant that it's clean and concise.

I tend towards the second option. What do you think? (Of course this would apply to all other usages of the trait as well.)

Reply:
I agree that the second option is clean and concise. The `use embedded_hal_async::i2c::I2c;` in the async tests would remain necessary as the call to i.e. `write(...)` must explicitly invoke the asynchronous version.